functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,332 @@
+
+#>   
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+
+    param(
+        [parameter(Mandatory = $true, ValueFromPipeline=$true)]
+		[object[]]$SqlServer,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$StepName,
+        [Parameter(Mandatory = $false)]
+        [string]$NewName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('ActiveScripting','AnalysisCommand','AnalysisQuery','CmdExec','Distribution','LogReader','Merge','PowerShell','QueueReader','Snapshot','Ssis','TransactSql')]
+        [string]$Subsystem,
+        [Parameter(Mandatory = $false)]
+        [string]$Command,
+        [Parameter(Mandatory = $false)]
+        [int]$CmdExecSuccessCode,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnSuccessAction,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($OnSuccessAction -ne 'GoToStep') -and ($_ -ge 1))
+            {
+                Throw "Parameter OnSuccessStepId can only be used with OnSuccessAction 'GoToStep'."
+            }
+            else 
+            {
+                $true    
+            }
+        })]
+        [int]$OnSuccessStepId,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnFailAction,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($OnFailAction -ne 'GoToStep') -and ($_ -ge 1))
+            {
+                Throw "Parameter OnFailStepId can only be used with OnFailAction 'GoToStep'."
+            }
+            else 
+            {
+                $true    
+            }
+        })]
+        [int]$OnFailStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseName,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseUserName,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryAttempts,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryInterval,
+        [Parameter(Mandatory = $false)]
+        [string]$OutputFileName,
+        [Parameter(Mandatory = $false)]
+        [string]$ProxyName,
+        [switch]$Silent
+    )
+
+    PROCESS
+    {
+        # Try connecting to the instance
+        Write-Message -Message "Attempting to connect to Sql Server.." -Level 2 -Silent $Silent
+        try 
+        {
+            $Server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential
+        }
+        catch 
+        {
+            Stop-Function -Message "Could not connect to Sql Server instance" -Silent $Silent -Target $SqlServer 
+            return
```

Comment:
As above, it might be nice to include the server you are trying to connect to. Also more details on why it failed:
`Stop-Function -Message "Could not connect to $instance : $($_.Exception.Message)" -Target $instance -InnerErrorRecord $_`
This way, the error text gets appended _and_ the error log will have the original exception, making it that much easier to troubleshoot. If you implement the loop, also use the `-Continue` parameter, to keep processing the next server if one failed.